utils/githubUtils.js

```diff
@@ -5,16 +5,16 @@ import {
 } from "./customErrors.js";
 import { GITHUB_TOKEN } from "../config/constants.js";
 
+// Initialize Octokit client with the GitHub token
+const octokit = github.getOctokit(GITHUB_TOKEN);
+
```

Comment:
Thinking that we should continue following a functional programming approach. When I created this code I opt an approach more oriented to isolation and self-containarization. But thinking that this client should be one for each user and the state has to be shared along the whole workflow. In that sense I will do the following:

1) Add function `createOctokitClient()` in gitHubUtils

```js
const createOctokitClient = (token) => {
  return github.getOctokit(token);
};
```
2) Modify functions `extractPullRequestData` and `createOrUpdateFile` to accept an oktokit parameter

```js
export const extractPullRequestData = async (octokit) => {
  // Existing logic...
};

export const createOrUpdateFile = async (
  octokit, owner, repo, path, content, message, branchRef
) => {
  // Existing logic...
};
```

3) Create the const variable `ocktokit` in `index.js` (main file that run workflow`

```js
const octokit = createOctokitClient(GITHUB_TOKEN);

// Use the functions with the dynamically created octokit instance
await extractPullRequestData(octokit);
await createOrUpdateFile(octokit, owner, repo, path, content, message, branchRef);
```